swap/swap.go

```diff
@@ -554,22 +570,20 @@ func (s *Swap) Deploy(ctx context.Context, initialDepositAmount uint64) (contrac
 
 // deployLoop repeatedly tries to deploy the swap contract .
 func (s *Swap) deployLoop(opts *bind.TransactOpts, defaultHarddepositTimeoutDuration time.Duration) (instance contract.Contract, err error) {
-	var tx *types.Transaction
 	for try := 0; try < deployRetries; try++ {
 		if try > 0 {
 			time.Sleep(deployDelay)
 		}
-		if instance, tx, err = contract.Deploy(opts, s.backend, s.owner.address, defaultHarddepositTimeoutDuration); err != nil {
-			auditLog.Warn("can't send chequebook deploy tx", "try", try, "error", err)
-			continue
-		}
-		if _, err := bind.WaitDeployed(opts.Context, s.backend, tx); err != nil {
+
+		chequebook, err := s.chequebookFactory.DeploySimpleSwap(opts, s.owner.address, big.NewInt(int64(defaultHarddepositTimeoutDuration)))
+		if err != nil {
 			auditLog.Warn("chequebook deploy error", "try", try, "error", err)
-			continue
+			return nil, err
```

Comment:
Does it still make sense here to actually do `deployRetries` here anymore? Isn't it the case that if it failed, we just return anyway and don't iterate anymore?

Reply:
Hm, I don't think i did that intentionally. I changed it back to continue.

That being said I'm not sure if the deploy loop is useful. What kind of error are we trying to handle here? The only situation I can think of where it would not work and then work a little latter was if the network connection to the eth node got cut (in that case it's unlikely it will work again after 1s too).

Reply:
Agreed about the deploy loop being redundant, I would propose to remove this, or add an issue so we won't forget to do this at a later point.